CRM/Activity/BAO/Activity.php

```diff
@@ -672,7 +672,7 @@ public static function getActivities($params, $getCount = FALSE) {
     $activities = array();
 
     // fetch all active activity types
-    $activityTypes = CRM_Core_OptionGroup::values('activity_type');
+    $activityTypes = CRM_Activity_BAO_Activity::buildOptions('activity_type_id', 'get');
```

Comment:
Implicit in the first line is $labelColumnName = 'label'
The use of get as context has the same effect

I'm finding the way this parameter is used in the function kinda odd - ie
```
    elseif (!empty($activityTypes) && count($activityTypes)) {
      $activityParams['activity_type_id'] = array('IN' => array_keys($activityTypes));
    }
```
seems to imply we want to exclude active ones here. I would be inclined to drop this from the PR to get the others through without this holding them up

Reply:
@eileenmcnaughton I've removed from this PR and raised a more complex change for this line in https://issues.civicrm.org/jira/browse/CRM-21111 - PR #10909 which should also improve performance of that function